src/tools/kata-ctl/src/check.rs

```diff
@@ -222,6 +232,72 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &str, parameter: &str) -> Result<String, String> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+    const MODULES_PATH: &str = "/sys/module";
+
+    // Partial check w/ modinfo
+    // verifies that the module exists
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module)
+        .output()
+        .expect("failed to execute process");
+
+    let status_modinfo = output_modinfo.status;
+
+    // Partial check w/ modprobe
+    // check that the module is already loaded
+    let output_modprobe = Command::new(MODPROBE_PATH)
+        .arg(MODPROBE_PARAMETERS_DRY_RUN)
+        .arg(MODPROBE_PARAMETERS_FIRST_TIME)
+        .arg(module)
+        .stdout(Stdio::piped())
+        .output()
+        .expect("failed to execute process");
+
+    let status_modprobe = output_modprobe.status;
+
+    // Here the current kernel parameter value is
+    // retrieved and returned on success
+    if status_modinfo.success() && !status_modprobe.success() {
+        let module_path = format!("{}/{}/parameters/{}", MODULES_PATH, module, parameter);
+
+        // retrieves the current parameter value
+        match read_file_contents(&module_path) {
+            Ok(result) => return Ok(result.replace('\n', "")),
+            Err(e) => return Err(e.to_string()),
+        }
+    }
+
+    // Catch any of the following situations and returns the corresponding error:
+    // 1. when a module is not found, or
+    // 2. when a module is not currently loaded.
+    match status_modprobe.code() {
+        Some(code) => {
+            // Module not found
+            if code != 0 {
+                let stderr = String::from_utf8_lossy(&output_modinfo.stderr);
+                let msg = format!("{}", stderr);
+                return Err(anyhow!(msg).to_string());
+            }
+
+            // Module not loaded
+            let msg = format!("Module not found: {}", module);
+            Err(anyhow!(msg).to_string())
```

Comment:
format! returns a string, do you need the to_string() call here and below?

Reply:
yeah you are right, I removed the `anyhow()` call, thanks @amshinde